tmt/steps/provision/__init__.py

```diff
@@ -1124,34 +1125,33 @@ def store_log(
             log_name: Optional[str] = None) -> None:
         """
         Save log content to a file.
+
         :param log_path: a path to save into,could be a directory
-        or a file path.
+            or a file path.
         :param log_content: content of the log.
         :param log_name: name of the log, if not set, log_path
-        is supposed to be '/dev/null' or  a file path.
+            is supposed to be a file path.
         """
-        # if log_path is file path or /dev/null
-        if not log_path.is_dir() or str(log_path) == '/dev/null':
+        # if log_path is file path
+        if not log_path.is_dir():
             log_path.write_text(log_content)
         # log_path is a directory
         else:
             if log_name:
-                name_str = log_name
+                (log_path / log_name).write_text(log_content)
             else:
-                name_str = 'tmt-guestlog-' + \
-                    datetime.datetime.now(datetime.timezone.utc).strftime("%Y%m%d%H%M%S")
-                self.warn(f'log_name is not set, using file name:{name_str}')
-            (log_path / name_str).write_text(log_content)
+                raise tmt.utils.GeneralError('log_name is None.')
 
     def handle_guest_logs(self,
                           log_path: Optional[Path] = None,
                           log_names: Optional[list[str]] = None) -> None:
         """
         Get log content and save it to a directory.
+
         :param log_path: a directory to save into.If not set,self.workdir
-        or Path.cwd() will be used.
+            or Path.cwd() will be used.
         :param log_names: name list of logs need to be handled.If not set,
-        self.log_names will be used.
+            self.log_names will be used.
```

Comment:
Few more notes:

* there should be a space between the end of a sentence and starting a new one, i.e. "a directory to save into. If not set...". The same applies to comas, i.e. "If not set, ..."
* "current working directory" is better than "Path.cwd()".
* "step's working directory" might be better than "self.workdir". It explains better the flow of decisions. You may use both, of course, but then you should take advantage of  Sphinx' domains: "If not set, step's working directory (:py:attr:\`workdir\`) or current working directory will be used."
* the same goes for "self.log_names" - "If not set, all guest logs will be collected." is better than "self.log_names" as it conveys the semantics of the parameter. A link to "self.log_names" might be nice, but it shouldn't be the first choice: "if not set, all guest logs would be collected, as reported by :py:attr:\`log_names\`."